File: python3.6/Ball.py
import math

class Ball:
    """
    Ball class
    Representation of a Ball
    """
    def __init__(self, x, y, r):
        self.x          = x
        self.y          = y
        self.r          = r
        self.d_x        = 0
        self.d_y        = 0
        self.speed      = 2
        self.direction  = 90
        self.color      = "#000000"

    def updateDeplacement(self):
        """
        Calcul of d_x and d_y for moving
        """
        self.d_x = self.speed * math.cos(self.direction)
        self.d_y = self.speed * math.sin(self.direction)

    def move(self):
        """
        Move the ball
        """
        self.x += self.d_x
        self.y += self.d_y

    def invertDx(self):
        """
        change the direction of the ball for x coord
        """
        self.d_x = -self.d_x

    def invertDy(self):
        """
        change the direction of the ball for y coord
        """
        self.d_y = -self.d_y

    def setColor(self, color):
        """
        set the color of the ball by the argument given
        """
        self.color = color

    def setSpeed(self, speed):
        """
        set the speed of the ball by the argument given
        """
        self.speed = speed

    def setDirection(self, d):
        """
        set the direction (in degree) of the ball by the argument given
        and update dx and dy
        """
        self.direction = d
        self.updateDeplacement()

    def setGraphics(self, graphics):
        """
        set the graphic (ellipse object) of the ball by the argument given
        """
        self.graphics = graphics

File: Java/Ball.java
package balls;

import java.awt.*;
import java.util.Random;

import javax.swing.*;

public class Ball extends JComponent {
	// Private args
	private int x,y,radius;
	private int d_x, d_y, speed = 2, direction = 90;
	private Color color = Color.red;
	
	public Ball(int p_x, int p_y, int p_radius) {
		x 		= p_x;
		y 		= p_y;
		radius		= p_radius;
		updateDeplacement();
	}
	
	public void invertDx() {
		d_x = -d_x;
	}
	
	public void invertDy() {
		d_y = -d_y;
	}
	
	public void setColor(Color c) {
		color = c;
	}
	
	public void setSpeed(int sp) {
		speed = sp;
	}
	
	public void setDirection(int d) {
		direction = d;
		updateDeplacement();
	}
	
	public void updateDeplacement() {
		d_x = (int) (speed * Math.cos(direction));
		d_y = (int) (speed * Math.sin(direction));
	}
	
	public void paint(Graphics g) {
		super.paint(g);
		g.setColor(color);
	   	g.fillOval(x,y,radius,radius);
	}
	
	public void move() {
		x += d_x;
		y += d_y;
	}
	
	public int getX() {
		return x;
	}
	
	public int getY() {
		return y;
	}
	
	public int getRadius() {
		return radius;
	}
	
	public int getDx() {
		return d_x;
	}
	
	public int getDy() {
		return d_y;
	}
	
	
	
}
